xstream/src/java/com/thoughtworks/xstream/converters/extended/OffsetDateTimeConverter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2017 XStream Committers.
+ * All rights reserved.
+ *
+ * The software in this package is published under the terms of the BSD
+ * style license a copy of which has been included with this distribution in
+ * the LICENSE.txt file.
+ *
+ * Created on 13. January 2017 by Matej Cimbora
+ */
+package com.thoughtworks.xstream.converters.extended;
+
+import com.thoughtworks.xstream.converters.ConversionException;
+import com.thoughtworks.xstream.converters.SingleValueConverter;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * Converts a {@link OffsetDateTime} to a string.
+ *
+ * @author Matej Cimbora
+ */
+public class OffsetDateTimeConverter implements SingleValueConverter {
+
+    private static final DateTimeFormatter FORMATTER;
+
+    static {
+        FORMATTER = new DateTimeFormatterBuilder()
+                .appendPattern("uuuu-MM-dd'T'HH:mm:ss")
```

Comment:
BUG - This formatter doesn't seem to include the offset, so instead of

2007-12-03T10:15:30+01:00

the text is

2007-12-03T09:15:30
or
2007-12-03T10:15:30

Either way, when this is read again, the offset of 1 hour is lost (and so either the time has changed or the timestamp has changed because it switched to offset 0).


Add a test to marshall and unmarshall an OffsetDateTime in a non UK-timezone (non-UTC/GMT timezone) to prove it works.